modules/table/src/main/java/org/apache/ignite/internal/table/distributed/command/CatalogVersionAware.java

```diff
@@ -25,4 +25,9 @@ public interface CatalogVersionAware {
      * Returns version that the Catalog must have locally for the node to be allowed to accept this command via replication.
      */
     int requiredCatalogVersion();
+
+    /**
+     * Setter for the required catalog version. Should be used with caution. Only object's creator should call it.
+     */
+    void requiredCatalogVersion(int version);
```

Comment:
Shouldn't `org.apache.ignite.internal.table.distributed.command.PartitionCommand#requiredCatalogVersion()` hide this setter?

Reply:
You mean moving it to subclass? Yes, I think I can do it, that's probably a good advice

Reply:
I change my opinion, this is the right place for this method. PartitionCommand might not be the only sub-interface of CatalogCersionAware in the future

Reply:
Ok